### ISSUE TITLE
YOLO-NAS with grayscale

### ISSUE THREAD

**Issue by Allison2324** [Deci-AI/super-gradients#1315]
Hello everyone, please give a guide on how to train a YOLO-NAS on a grayscale dataset, and how to submit data to a dataloader without errors.

I changed in_channels to 1 and when I tried to solve it - I get problems like these:
1) Dataloader was reading images with 3 channels
2) After 1, I tried to use torchvision transform Grayscale, but it didn't work
3) When I edited Grayscale transform by the example of other SG transforms - was a problem with the types

So, maybe someone knows how to solve this problem?

**Comment by AkashParua** [comment]
``Here is an example using grayscale images : [https://learnopencv.com/train-yolo-nas-on-custom-dataset/](url) .
I think the model needs 3 channel images . You can change your images into 3 channel images and then use them for training -
`import cv2

def convert_grayscale_to_3channel(input_path, output_path):
    # Read the grayscale image
    grayscale_image = cv2.imread(input_path, cv2.IMREAD_GRAYSCALE)

    # Duplicate the grayscale image into three channels
    duplicate_image = cv2.merge((grayscale_image,) * 3)

    # Save the duplicate 3-channel image
    cv2.imwrite(output_path, duplicate_image)

if __name__ == "__main__":
    input_image_path = "path_to_grayscale_image.png"
    output_image_path = "path_to_output_3channel_image.png"

    convert_grayscale_to_3channel(input_image_path, output_image_path)
`

**Comment by BloodAxe** [comment]
Truth is we haven't tested this use case so indeed there may be some mismatches / unsupported pieces of code that are hard-coded for three-channel input.

So yes, as of now the easiest solution is to convert dataset to monochrome 3-channel format. 

We may improve the support of single channel data in future but as of now I'm not ready to commit to specific timeline when to expect this to happen
